src/Main.scala

```diff
@@ -0,0 +1,38 @@
+import java.util.concurrent.{ExecutorService, Executors, PriorityBlockingQueue}
+
+import scala.concurrent.ExecutionContext
+import scala.util.Try
+
+object Main {
+
+  object TimeOrder extends Ordering[ScheduledTask[_]] {
+    override def compare(x: ScheduledTask[_], y: ScheduledTask[_]): Int = x.timestamp compare y.timestamp
+  }
+
+  def main(args: Array[String]): Unit = {
+
+    // ExecutionContext for Scheduler
+    implicit val ec = ExecutionContext.fromExecutorService(Executors.newFixedThreadPool(5))
+
+    val size: Int = Try(args(0)).map(_.toInt).getOrElse(100)
+    // Java's PriorityQueue is MIN heap by default
```

Comment:
Maybe if it's a queue of integers, but for objects I think you'll always need to define the ordering.